XenModel/Actions/Updates/DownloadAndUnzipXenServerPatchAction.cs

```diff
@@ -68,7 +70,9 @@ public DownloadAndUnzipXenServerPatchAction(string patchName, Uri uri, string ou
             updateName = patchName;
             address = uri;
             downloadUpdate = address != null;
-            zippedFileName = outputFileName;
+            nonExtractUpdate = downloadUpdate &&
+                               Array.Exists(updateFileExtensions, item => address.ToString().Contains(item));

```

Comment:
If an URL contains an extension, it must be a file that doesn't need to be extracted. Am I right?
I think we'd better guarantee the URL contains "iso" (for example) as an extension (at the end). If the URL is like "xxxxxisoxxxx", it also pass the check but it was not what we want.
Please correct me if my understanding is incorrect.